db/migrate/20220825115850_convert_chp_heat_inputs_to_electricity.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'etengine/scenario_migration'
+
+# Updates various industry capacity input values to set electricity output rather than heat output.
+#
+# See https://github.com/quintel/etsource/issues/2716
+class ConvertChpHeatInputsToElectricity < ActiveRecord::Migration[7.0]
+  include ETEngine::ScenarioMigration
+
+  INPUTS = {
+    capacity_of_industry_chp_combined_cycle_gas_power_fuelmix:
+      :industry_chp_combined_cycle_gas_power_fuelmix,
+    capacity_of_industry_chp_engine_gas_power_fuelmix:
+      :industry_chp_engine_gas_power_fuelmix,
+    capacity_of_industry_chp_turbine_gas_power_fuelmix:
+      :industry_chp_turbine_gas_power_fuelmix,
+    capacity_of_industry_chp_ultra_supercritical_coal:
+      :industry_chp_ultra_supercritical_coal,
+    capacity_of_industry_chp_wood_pellets:
+      :industry_chp_wood_pellets
+  }
+
+  def up
+    migrate_scenarios do |scenario|
+      INPUTS.each do |input, node|
+        begin
+          next if !scenario.user_values.key?(input) || scenario.user_values[input].zero?
+        rescue Psych::DisallowedClass
+          # Ignore older scenarios with illegal values.
+          next
+        end
```

Comment:
The migration failed for me on a copy of the beta database:

```
Caused by:
NoMethodError: undefined method `zero?' for "":String

          next if !scenario.user_values.key?(input) || scenario.user_values[input].zero?
                                                                                  ^^^^^^
```
We could for instance add the `NoMethodError` to the `rescue` block, as this sounds like another old scenario with illegal values.

```suggestion
        begin
          next if !scenario.user_values.key?(input) || scenario.user_values[input].zero?
        rescue Psych::DisallowedClass, NoMethodError
          # Ignore older scenarios with illegal values.
          next
        end
```

Reply:
Thanks for catching this. I tested with a copy of the production DB, but it does indeed appear that one beta scenario has an empty value for an input. I think it's okay to skip the scenario.

I've adjusted the PR to account for this.